### ISSUE TITLE
[BUG] package-lock.json double-spaced when package.json has an empty line after the opening brace

### ISSUE THREAD

**Issue by MattF-NSIDC** [npm/cli#5622]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When running `npm install`, the `package-lock.json` output is double-spaced when a blank line exists between the opening brace and the first key.

This JSON produces a double-spaced lockfile:

```json
{

  "name": "test-doublespace",
  "dependencies": {
    "left-pad": "latest"
  }
}
```

and this one doesn't:

```json
{
  "name": "test-doublespace",
  "dependencies": {
    "left-pad": "latest"
  }
}
```

I've tried putting blank lines in other places, but have only been able to reproduce it with a blank line after the opening brace.

### Expected Behavior

Blank lines are valid in JSON and are not considered to be "data", so I would expect any blank lines to have no impact on the locked data.

### Steps To Reproduce

1. Create a new empty directory
2. Create a test `package.json
    ```
    echo '{
    
      "name": "test-doublespace",
      "dependencies": {
        "left-pad": "latest"
      }
    }' > package.json
    ```
3. Run `npm install`
4. View `package-lock.json`. It should be double-spaced!

### Environment

- npm: 8.19.2
- Node.js: v16.17.1
- OS Name: Ubuntu 20.04
- System Model Name: Lemur Pro
- npm config:
```ini
; "user" config from /home/mfisher/.npmrc

//registry.npmjs.org/:_authToken = (protected) 

; node bin location = /home/mfisher/.nvm/versions/node/v16.17.1/bin/node
; node version = v16.17.1
; npm local prefix = /tmp/foo
; npm version = 8.19.2
; cwd = /tmp/foo
; HOME = /home/mfisher
; Run `npm config ls -l` to show all defaults.
```

**Comment by MattF-NSIDC** [comment]
There is something about the simplicity of reproducing and the weirdness of the behavior that made this special. This added a hard-to-explain amount of fun to my work day. Thank you for the gift of this bug :bow:

**Comment by ljharb** [comment]
It seems like it's not really a bug - it's picking up the first-used empty-line formatting from the package.json and replicating it in the lockfile.

**Comment by MattF-NSIDC** [comment]
Is that intended? Are there other things NPM will do to "transfer" formatting from `package.json` to `package-lock.json`? I've tried these `package.json`s and they all produce the same lockfile (the example in the OP being the exception). My expectation was that formatting of `package.json` doesn't matter.

```
{
  "name": "test-doublespace",
  "dependencies": {
    "left-pad": "latest"
  }
}
```

```
{ "name": "test-doublespace",
  "dependencies": {
    "left-pad": "latest"
  }
}
```

```
{"name": "test-doublespace", "dependencies": {"left-pad": "latest"}}
```

**Comment by ljharb** [comment]
Yes, npm copies "is there a trailing newline" and also "indentation" - ie, tabs are preserved, as is the number of spaces you used.

**Comment by MattF-NSIDC** [comment]
The indentation behavior is intuitive to me, but replicating the newlines certainly wasn't. Is the replication of newlines a new feature in NPM v8? v7 "transfers" indentation from `package.json` to `package-lock.json`, but does not broadcast blank lines like v8 does.

**Comment by ljharb** [comment]
I don't claim to understand it entirely either, I just assume it's an artifact of the normal formatting preservation behavior.

**Comment by lukekarrys** [comment]
I believe the culprit here is [our json parse package](https://github.com/npm/json-parse-even-better-errors), specifically [these lines](https://github.com/npm/json-parse-even-better-errors/blob/37b98ee487dfa92f11ee4d9a0a3f15f79ff15b01/lib/index.js#L71-L78).

This behavior looks like an unintended side effect of respecting indentation and newlines, but I think a fix for not double (or triple) spacing these lines would be nice.
